lib/rules/no-useless-assignment.js

```diff
@@ -555,6 +555,26 @@ module.exports = {
                         }
                     }
 
+                    // Check if the variable is used in a catch clause
+                    const isUsedInCatchClause = variable.references.some(reference => {
+                        let currentNode = reference.identifier.parent;
+
+                        if (currentNode.type !== "AssignmentExpression") {
+                            while (currentNode) {
+                                if (currentNode.type === "CatchClause") {
+                                    return true;
+                                }
+                                currentNode = currentNode.parent;
+                            }
```

Comment:
This search up to the root increases the timing for this rule by around 20% when linting this project. Can we maybe store `CatchClause` nodes in `scopeStack`, and at some later point check whether a reference is in the range of some of them?

Reply:
Can you push the changes with a test case that doesn't work due to this so I can take a look?

Reply:
Catch clauses are stored when they're traversed, but we're checking them before they're traversed, so we'd need to move the check somewhere inside `function verify`.